test/module/irohad/synchronizer/synchronizer_test.cpp

```diff
@@ -356,12 +360,30 @@ TEST_F(SynchronizerTest, ExactlyThreeRetrievals) {
   ASSERT_TRUE(wrapper.validate());
 }
 
+MATCHER_P(StringEqSharedPtr, ptr, "equals " + *ptr) {
+  return std::string_view{arg} == std::string_view{*ptr};
+}
+
+template <typename Strong>
+struct StringViewHelper {
+  std::shared_ptr<std::string> holder{std::make_shared<std::string>()};
```

Comment:
```suggestion
  std::string holder;
```
Is there a need for a shared pointer? Also, it is possible to assign string view to a string, and they both have equality comparison

Reply:
string is shared with the matcher object in call expectations